docs/docs/adding-page-transitions-with-plugin-transition-link.md

```diff
@@ -0,0 +1,177 @@
+
+```javascript
+<TransitionLink
+  exit={{
+    length: length,
+    // highlight-next-line
+    trigger: ({ exit, node }) =>
+      this.someCustomDefinedAnimation({ exit, node, direction: "out" }),
+  }}
+  entry={{
+    length: 0,
+    // highlight-next-line
+    trigger: ({ exit, node }) =>
+      this.someCustomDefinedAnimation({ exit, node, direction: "in" }),
+  }}
+  {...props}
+>
+  {props.children}
+</TransitionLink>
+```
+
+#### Using passed props
+
+The exiting and entering pages/templates involved in the transition will receive props indicating the current transition status, as well as the `exit` or `enter` props defined on the `TransitionLink`.
+
+```javascript
+const PageOrTemplate = ({ children, transitionStatus, entry, exit }) => {
+  console.log(transitionStatus, entry, exit)
+  return <div className={transitionStatus}>{children}</div>
+}
+```
+
+You can combine these props with a _declarative_ state-based animation libraries like [react-pose](https://popmotion.io/pose/) or [react-spring](http://react-spring.surge.sh/) to specify transitions for exiting and entering a page.
+
+If you want to access these props in one of your components instead of a page/template, you should wrap your component in the `TransitionState` component. This component takes a function that will have access to the same props as above, which you can then use in your component.
+
+Here's an example using `TransitionState` and `react-pose` to trigger enter/exit transitions for a `Box` component.
+
+```javascript
+import { TransitionLink } from "gatsby-plugin-transition-link"
+
+const Box = posed.div({
+  hidden: { opacity: 0 },
+  visible: { opacity: 1 },
+})
+
+<TransitionState>
+      {({ transitionStatus, exit, enter }) => {
+        console.log('exit object is', exit)
+        console.log('enter object is', enter)
+
+        return (
+            <Box
+              className="box"
+              pose={
+                ['entering', 'entered'].includes(transitionStatus)
+                  ? 'visible'
+                  : 'hidden'
+              }
+            />
+        )
+      }}
+</TransitionState>
+```
+
+Now, the `Box` component will be aware of the transition status of the page it's a child of, and it will fade in/out accordingly.
+
+## Excluding elements from page transitions
+
+You may want to have elements on a page that persist throughout the page transition (_ex. a site-wide header_). This can be accomplished by wrapping elements in the `TransitionPortal` component.
+
+```javascript
+import { TransitionPortal } from "gatsby-plugin-transition-link"
+```
+
+```javascript
+<TransitionPortal>
+  <SomeComponent>
+    This component will sit on top of both pages, and persist through page
+    transitions.
+  </SomeComponent>
+</TransitionPortal>
+```
+
+As always, check out [the docs](https://transitionlink.tylerbarnes.ca/docs/transitionportal/) for more information about `TransitionPortal`.
```

Comment:
This link would be more descriptive in preparing screen reader users for where it goes by tweaking it a bit:

"check out [the TransitionPortal docs](https://...) for more information"